src/Passes/Source/Commandline/ConfigurationBindInterface.hpp

```diff
@@ -0,0 +1,92 @@
+#pragma once
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Commandline/ParameterParser.hpp"
+
+#include "Llvm/Llvm.hpp"
+
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+#include <type_traits>
+#include <typeindex>
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    /// Interface class to bind a variable to a configuration flag. This class provides
+    /// the neccessary means to
+    class IConfigBind
+    {
+      public:
+        using String = std::string;
+
+        // Deteled constructors, deleted operators and destructor
+        //
+        IConfigBind(IConfigBind const&) = delete;
+        IConfigBind(IConfigBind&&)      = delete;
+        IConfigBind& operator=(IConfigBind const&) = delete;
+        IConfigBind& operator=(IConfigBind&&) = delete;
+        virtual ~IConfigBind();
+
+        // Interface
+        //
+
+        /// Interface function to register configuration in the parser. This function
+        /// register the configuration to the parameter parser. This makes the configuration
+        /// available in the parameter parsers help function.
+        virtual bool setupArguments(ParameterParser& parser) = 0;
+
+        /// Interface function to extract configuration. Given an instance of the
+        /// parameter parser, the this function is meant to extract and update the
+        /// bound variable value if present.
+        virtual bool configure(ParameterParser const& parser) = 0;
```

Comment:
1. Consider describing the return value.
2. 
```suggestion
        /// Interface function to register configuration in the parser. This function
        /// register the configuration to the parameter parser. This makes the configuration
        /// available in the parameter parsers help function.
        virtual bool setupArguments(ParameterParser& parser) = 0;

        /// Interface function to extract configuration. Given an instance of the
        /// parameter parser, this function is meant to extract and update the
        /// bound variable value if present.
        virtual bool configure(ParameterParser const& parser) = 0;
```
3. The comment is somewhat vague:
```c++
        /// Interface function to extract configuration. Given an instance of the
        /// parameter parser, the this function is meant to extract and update the
        /// bound variable value if present.
```
> to extract configuration

To extract how? Where to save after the extraction?
> and update the bound variable value

Update how? What is bound variable?